### DISCUSSION TITLE
Is there a way to get rid of NextJS image inline styles for better SEO?

### DISCUSSION THREAD

**Discussion by Vipse** [vercel/next.js#45209] [Help] [answered]
### Summary

NextJS image adds inline styles to the img tag 
it can be either `style="color:transparent"` or `style="position:absolute;height:100%;width:100%;left:0;top:0;right:0;bottom:0;color:transparent"` with 'fill' prop

**Comment by leerob** [comment, marked as answer]
Inline styles shouldn't affect SEO – these styles are needed for the component to work properly.

**Reply by vzrenggamani** [reply]
Hi, I have same problems here @leerob, just a little, the CSP are showing inline-styles are bad and the `next/image` are returning `inline-styles` by default, any chances I could do to take that off?

Thank you so for the works so far

**Reply by borisdiakur** [reply]
➡️ https://github.com/vercel/next.js/issues/45184

**Reply by msmi1433** [reply]
@vzrenggamani did you get anyhwere with this. I'm having the same issue with CSP - think this line is violating my header even though I have implemented the nonce middleware as described [here](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy).

**Reply by jordanpurinton** [reply]
This is also an issue for me

**Reply by jeanvcastro** [reply]
Has anyone been able to solve a CSP issue?

**Reply by ph1p** [reply]
In addition to the CSP, there are also accessibility tools that display errors and point out the following rule https://www.w3.org/TR/WCAG20-TECHS/F24.html.

**Reply by psztoch** [reply]
`color: transparent` is very bad for svg with `currentColor`.

**Reply by David-Slack** [reply]
I've had to do use color: #000 !important;. Horrible solution. Really fighting with version 15 of Next

**Comment by mabelzanecali** [comment]
If you're looking to optimize your Next.js setup, you must use strategies like Incremental Static Regeneration (ISR) and set proper Cache-Control headers. For expert help, you must use [Pearl Lemon Clutch reviews](https://clutch.co/profile/pearl-lemon#highlights) they have a proven track record in web performance optimization.
